docs/Design/SoftDetailedDes/MIS.tex

```diff
@@ -625,6 +625,101 @@ \subsubsection{Access Routine Semantics}
 \end{itemize}
 
 
+\subsubsection{Local Functions}
+None
+
+\newpage
+
+\section{MIS of User Analytics Module} \label{mLocation} 
+The User Analytics Module provides analytics and feedback on user purchasing behaviour, offering insight on recent and future purchasing for individual users.
+
+\subsection{Module}
+
+userAnalytics
+
+\subsection{Uses}
+
+Database Driver Module (Section \ref{mDBDriver})
+Recommendation Engine (Section \ref{mRec})
+
+\subsection{Syntax}
+
+\subsubsection{Exported Constants}
+None
+
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+\begin{tabular}{p{5cm} p{3cm} p{4cm} p{4cm}}
+\hline
+\textbf{Name} & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+\hline
+getSpendingByCategory & $\mathbb{N}^2$ & List$<$(String, List$<$Item$>$)$>$  & - \\
+getSpendingByTime & $\mathbb{N}^2$ & List$<$(String, $\mathbb{R}$+)$>$ & - \\
```

Comment:
For the Users module, Sawyer used DateTime as his type for Dates, but I assume you're using Unix timestamps. We should probably decide on a consistent type. I might also have to make changes to mine too depending on what we're saying we're using for date time.

Reply:
You're right but it's probably fine... very easy to convert in and out, can change specs later too